test/e2e/framework/util.go

```diff
@@ -4989,6 +4989,25 @@ func (f *Framework) NewTestPod(name string, requests v1.ResourceList, limits v1.
 	}
 }
 
+// NewAgnhostPod returns a pod that uses the agnhost image. The image's binary supports various subcommands
+// that behave the same, no matter the underlying OS.
+func (f *Framework) NewAgnhostPod(name string, args ...string) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "agnhost",
+					Image: imageutils.GetE2EImage(imageutils.Agnhost),
```

Comment:
just to note that long term we need to decouple the framework from the parent folder (test).
this utility function seems fine for now, but are there going to be other consumers of this image apart from `It("should support configurable pod DNS....`?

Reply:
Keeping in mind that this image is supposed to be extendable and used whenever we have significant differences between Windows and Linux (especially commands), yes.